addons/block_code/ui/blocks/block/block.gd

```diff
@@ -27,7 +27,16 @@ signal modified
 @export var scope: String = ""
 
 ## The resource containing the definition of the block
-@export var definition: BlockDefinition
+@export var definition: BlockDefinition:
+	set(value):
+		var is_changed := definition != value
+		if definition and is_changed:
+			definition.changed.disconnect(_on_definition_changed)
```

Comment:
The amount of times that I wanted to hook to the changed signal of custom resources. Thanks!